src/composables/secureStorageRef.ts

```diff
@@ -107,19 +107,18 @@ export function useSecureStorageRef<T = string | object | any[]>(
 
     /**
      * Clear the state when the user logs out.
-     * This ensures that the state is not leaked if someone removes the modal from the DOM.
+     * Ensures that the state is not leaked if an attacker removes the login modal from the DOM.
      */
-    const { encryptionData } = useAccounts();
-    watch(encryptionData, async (val) => {
-      if (IS_MOBILE_APP) {
-        return;
-      }
-      if (!val) {
-        decryptedState.value = initialState;
-      } else {
-        decryptedState.value = await getDecryptedValue(innerState.value);
-      }
-    }, { immediate: true, deep: true });
+    if (!IS_MOBILE_APP) {
+      const { encryptionKey } = useAccounts();
+      watch(encryptionKey, async (key) => {
+        if (!key) {
+          decryptedState.value = initialState;
+        } else {
+          decryptedState.value = await getDecryptedValue(encryptedState.value);
+        }
+      }, { immediate: true, deep: true });
```

Comment:
I suggest to consider using asynchronous computed property (https://vueuse.org/core/computedAsync/) instead of watcher as the watcher exists only to sync one value with another.